---
title: "Clustering and taxonomic classification of uncultivated viral genomes"
teaching: 0
exercises: 15
questions:
- "What are we going to do today?"
objectives:
keypoints:
---
Viral taxonomy is a very active field. As you know by now, the number of viral sequences is increasing rapidly in multiple databases, reflecting the discovery of uncultivated viruses with metagenomics as well as other methods. The fact that similar sequences are found in different places indicates that the virus is really replicating. Some of these sequences can be directly placed into the viral taxonomy at the species or genus rank, for taxa that are described with the [International Committee on Taxonomy of Viruses (ICTV)](https://talk.ictvonline.org/). In case new viral lineages are discovered, ICTV hopes that researchers will submit taxonomy proposals to register the lineage as an official taxonomic group. Specific properties of this group can be described, such as the set of hallmark genes that characterises the [*Schitoviridae*](https://pubmed.ncbi.nlm.nih.gov/33008130/). This bookkeeping is important so that virologists and viral ecologists with a new viral sequence know what they are dealing with, and what the likely properties of the virus are. Currently there is no single best method (or consensus) to classify all viruses: methods differ depending on the taxonomic rank (say, family or species), but they can also vary among different types of viruses.

Today we will use different approaches to try to taxonomically annotate the viral sequences you have recovered in the previous days.

> ## More info:
>For more information on taxonomic classification you can read the [Perspective on taxonomic classification of uncultivated viruses (Dutilh *et al* 2021). ](https://www.sciencedirect.com/science/article/pii/S1879625721001334)
{: .callout}


{% include links.md %}
